This patch fixes `eslint-plugin-react` to be compatible with ESLint 10.
We're patching because seems like it'll take a while to merge.
Patch copied from https://github.com/jsx-eslint/eslint-plugin-react/pull/3979

diff --git a/lib/rules/forward-ref-uses-ref.js b/lib/rules/forward-ref-uses-ref.js
index 3a0b7de4c951e7ea92a9de7bac924ba95d643342..3a02572c3ace22afe0b589d5cdd03fb9ea889c99 100644
--- a/lib/rules/forward-ref-uses-ref.js
+++ b/lib/rules/forward-ref-uses-ref.js
@@ -6,6 +6,7 @@
 
 const isParenthesized = require('../util/ast').isParenthesized;
 const docsUrl = require('../util/docsUrl');
+const getSourceCode = require('../util/eslint').getSourceCode;
 const report = require('../util/report');
 const getMessageData = require('../util/message');
 
@@ -57,7 +58,7 @@ module.exports = {
   },
 
   create(context) {
-    const sourceCode = context.getSourceCode();
+    const sourceCode = getSourceCode(context);
 
     return {
       'FunctionExpression, ArrowFunctionExpression'(node) {
diff --git a/lib/rules/jsx-filename-extension.js b/lib/rules/jsx-filename-extension.js
index 4a2cfeede66b68255ca48bc38c0eb2f1537a785a..0e5f02574dea09db134f599a34e0c678f2a077f4 100644
--- a/lib/rules/jsx-filename-extension.js
+++ b/lib/rules/jsx-filename-extension.js
@@ -7,6 +7,7 @@
 
 const path = require('path');
 const docsUrl = require('../util/docsUrl');
+const getFilename = require('../util/eslint').getFilename;
 const report = require('../util/report');
 
 // ------------------------------------------------------------------------------
@@ -61,7 +62,7 @@ module.exports = {
   },
 
   create(context) {
-    const filename = context.getFilename();
+    const filename = getFilename(context);
 
     let jsxNode;
 
diff --git a/lib/util/eslint.js b/lib/util/eslint.js
index 79a0537f3b4f52b04b29d46e89ce00f6da8fe579..bcf9186abf87ed636ffb30578bcda94a792249cf 100644
--- a/lib/util/eslint.js
+++ b/lib/util/eslint.js
@@ -4,6 +4,10 @@ function getSourceCode(context) {
   return context.getSourceCode ? context.getSourceCode() : context.sourceCode;
 }
 
+function getFilename(context) {
+  return context.getFilename ? context.getFilename() : context.filename;
+}
+
 function getAncestors(context, node) {
   const sourceCode = getSourceCode(context);
   return sourceCode.getAncestors ? sourceCode.getAncestors(node) : context.getAncestors();
@@ -38,6 +42,7 @@ function getText(context) {
 
 module.exports = {
   getAncestors,
+  getFilename,
   getFirstTokens,
   getScope,
   getSourceCode,
diff --git a/lib/util/makeNoMethodSetStateRule.js b/lib/util/makeNoMethodSetStateRule.js
index b6c55d5cf61fac650fb67b1a91d037baf4992e36..a6a7b7c6847c135c12c0d547f9347c5ce2bc5a3e 100644
--- a/lib/util/makeNoMethodSetStateRule.js
+++ b/lib/util/makeNoMethodSetStateRule.js
@@ -51,14 +51,16 @@ function shouldBeNoop(context, methodName) {
  * @returns {import('eslint').Rule.RuleModule}
  */
 module.exports = function makeNoMethodSetStateRule(methodName, shouldCheckUnsafeCb) {
+  const docs = {
+    description: `Disallow usage of setState in ${methodName}`,
+    category: 'Best Practices',
+    recommended: false,
+    url: docsUrl(mapTitle(methodName)),
+  };
+
   return {
     meta: {
-      docs: {
-        description: `Disallow usage of setState in ${methodName}`,
-        category: 'Best Practices',
-        recommended: false,
-        url: docsUrl(mapTitle(methodName)),
-      },
+      docs,
 
       messages,
 
diff --git a/lib/util/version.js b/lib/util/version.js
index c9dc683b7d149c5a3a941bdcbd32f657facdfbf6..2bfbf2ecd751c2ef83df7738d70026321d777d8a 100644
--- a/lib/util/version.js
+++ b/lib/util/version.js
@@ -11,6 +11,7 @@ const path = require('path');
 const resolve = require('resolve');
 const semver = require('semver');
 const error = require('./error');
+const getFilename = require('./eslint').getFilename;
 
 const ULTIMATE_LATEST_SEMVER = '999.999.999';
 
@@ -28,7 +29,7 @@ function resetDetectedVersion() {
 
 function resolveBasedir(contextOrFilename) {
   if (contextOrFilename) {
-    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : contextOrFilename.getFilename();
+    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : getFilename(contextOrFilename);
     const dirname = path.dirname(filename);
     try {
       if (fs.statSync(filename).isFile()) {
diff --git a/types/eslint-core-augment.d.ts b/types/eslint-core-augment.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..3abb08c3fd7abfd02ea3dd487df96e3aa80faeda
--- /dev/null
+++ b/types/eslint-core-augment.d.ts
@@ -0,0 +1,13 @@
+import '@eslint/core';
+
+declare module '@eslint/core' {
+  interface SettingsConfig {
+    react?: {
+      pragma?: string;
+      fragment?: string;
+      createClass?: string;
+    };
+  }
+}
+
+export {};
